src/decode.rs

```diff
@@ -86,7 +86,7 @@ impl<'a, I: AsRef<[u8]>> DecodeBuilder<'a, I> {
     /// ```
     ///
     pub fn into_vec(self) -> Result<Vec<u8>, DecodeError> {
-        let mut output = vec![0; (self.input.as_ref().len() / 8 + 1) * 6];
+        let mut output = vec![0; std::cmp::max(self.input.as_ref().len(), (self.input.as_ref().len() / 8 + 1) * 6)];
```

Comment:
```suggestion
        let mut output = vec![0; self.input.as_ref().len()];
```

The decoded length is guaranteed to be at most the encoded length (for this worst case of all zeros). There's no need to calculate the estimated length since that's going to be shorter (other than the cases where it was already over-estimating the length).